stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -152,35 +146,33 @@ def handle_api_exception(self, code=None, response_txt=''):
         ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
         return return_obj
 
-    def handle_metadata_and_page_size(self, offset, length, metadata):
+    def handle_metadata(self, offset, length, metadata):
         """
-        Processing metadata information and page size
+        Processing metadata information
         :param offset, int
         :param length, int
         :param metadata, dict
-        :return: return_obj, dict
+        :return: total_records, int
+                 page_number, int
+                 offset, int
         """
-        result_count = 0
         page_number = 1
 
+        # calculating the overall record count
+        total_records = offset + length
+        if self.api_client.result_limit < total_records:
+            total_records = self.api_client.result_limit
+
         if metadata:
             if isinstance(metadata, dict) and metadata.get('page_number'):
-                result_count = int(metadata.get('result_count', 0))
                 page_number = int(metadata.get('page_number', 1))
+                # overwrite the offset for current batch.
+                offset = abs(offset - ((page_number - 1) * self.api_client.api_page_size))
```

Comment:
Can you expand on this? I'm not really sure what this does.

If my offset is 0, this becomes larger over iterations. If my offset is 4,000, this becomes smaller over time and eventually starts to increase.

Reply:
offset passed from the function parameter is not directly used for slicing.
after calculating offset for the current batch, It will be used for slicing.

if the offset is 0 page_number will start from 1.

if the offset is 4000 page_number will come as 5 from the metadata. 
It will not start form 1 because already 4 pages are processed.

example 1: offset starts from 0.
for the first batch
offset=0 length=2000 pagesize=1000 page_number=1
offset = abs(0 - ((1 - 1) * 1000))
offset for the current batch is 0.

for the second batch
offset=2000 length=2000 pagesize=1000 page_number=3
offset = abs(2000 - ((3 - 1) * 1000))
offset for the current batch is 0.

for the third batch
offset=4000 length=2000 pagesize=1000 page_number=5
offset = abs(4000 - ((5 - 1) * 1000))
offset for the current batch is 0.


example 2: offset not starts from 0.
for the first batch
offset=500 length=2000 pagesize=1000 pagenumber=1
offset = abs(500 - ((1 - 1) * 1000))
offset for the current batch is 500.

for the second batch
offset=2500 length=2000 pagesize=1000 pagenumber=3
offset = abs(2500 - ((3 - 1) * 1000))
offset for the current batch is 500.

Reply:
Can you provide some context on why this needs to be done? What happens if my offset is less than the page_number * pagesize?

Reply:
Updated the pagination as per the suggestion.